go/libraries/doltcore/schema/alterschema/droppk_test.go

```diff
@@ -124,3 +124,134 @@ func TestDropPk(t *testing.T) {
+				"create table child (id int, name varchar(1), age int, primary key (id), constraint `fk` foreign key (id) references parent (id))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
+		},
+		{
+			name: "no error if backup index for single FK on compound pk drop",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id, age), key `backup` (age))",
+				"create table child (id int, name varchar(1), age int, primary key (id), constraint `fk` foreign key (age) references parent (age))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
+		},
+		{
+			name: "no error if compound backup index for compound FK",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id, age), key `backup` (id, age))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (id, age) references parent (id, age))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
+		},
+		{
+			name: "no error if both invalid and valid backup index",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id, age), key `bad_backup1` (id, age), key `bad_backup2` (id), key `backup` (age, id))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (age) references parent (age))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
+		},
+		{
+			name: "error if FK ref but no backup index for single pk",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id))",
+				"create table child (id int, name varchar(1), age int, primary key (id), constraint `fk` foreign key (id) references parent (id))",
+			},
+			exit:      1,
+			fkIdxName: "id",
+		},
+		{
+			name: "error if FK ref but bad backup index",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id), key `bad_backup2` (age))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (id) references parent (id))",
+			},
+			exit:      1,
+			fkIdxName: "id",
+		},
+		{
+			name: "error if misordered compound backup index for FK",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, constraint `primary` primary key (id), key `backup` (age, id))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (id) references parent (id))",
+			},
+			exit:      1,
+			fkIdxName: "id",
+		},
+		{
+			name: "error if incomplete compound backup index for FK",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, constraint `primary` primary key (age, id), key `backup` (age, name))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (age, id) references parent (age,  id))",
+			},
+			exit:      1,
+			fkIdxName: "ageid",
+		},
+	}
```

Comment:
Is there any priority or a specific order that governs which index is chosen? Now we don't allow for indexes to cover the same exact set of keys, but you can declare composite indexes that are partial keys of other indexes. Is the smallest index always chosen? The index that was declared first? A unique index takes priority over all others if one is available and the data doesn't have any duplicate rows? Some combination of the aforementioned?

We don't have to implement all of these (and can't for some like index declaration order since we don't preserve that information), but it's good to know, and perhaps even drop a comment in the source. But for something like unique key preference, I think we should definitely implement that.